samples/msal-node-samples/custom-INetworkModule-and-network-tracing/HttpClient.ts

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    INetworkModule,
+    NetworkRequestOptions,
+    NetworkResponse
+} from "@azure/msal-common";
+import http from "http";
+import https from "https";
+
+/**
+ * This class implements the API for network requests.
+ */
+export default class HttpClient implements INetworkModule {
+
+    /**
+     * Http Get request
+     * @param url
+     * @param options
+     */
+    async sendGetRequestAsync<T>(
+        url: string,
+        options?: NetworkRequestOptions,
```

Comment:
Is `NetworkRequestOptions` meant to be a public class? Shuld we expose only the proxy setting?

Reply:
I copied this HttpClient directly from https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/src/network/HttpClient.ts and shouldn't be making any changes to it here.